### ISSUE TITLE
Our application of patch for `oauth-0.9.15` fails on CI when `oauth-0.10` is installed by npm

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#1724]
As described in this Discord convo https://discord.com/channels/686873244791210014/1204411999795155034/1204411999795155034 ,
recently one of our npm deps for `server` code started pulling in `oauth-0.10.0` instead of `oauth-0.9.15`. This wouldn't be a problem if we didn't have patch specifically for `oauth-0.9.15` (you can find it in `waspc/data/Generator/templates/server/patches/oauth+0.9.15.patch`). When `patch-package` tries to apply it to the code (that happens because we have it under `"postinstall"` in package.json`), it errors out on CI because it detects that version of `oauth` installed doesn't match the package and finds that potentially problematic, wants us to fix that. This doesn't happen outside of CI because that is how `patch-package` works -> in CI it fails on such issues, outside of CI it just warns you.

while I proposed a quick fix in the Discord convo, we should fix this properly.
We should fix it both for 0.11.8 and 0.12 versions of Wasp.

Possible solutions:
1. Somehow ensure version of `oauth` is always 0.10 from now on. Maybe recognize which dep in our server/package.json started pulling in `oauth` 0.10 and update it significantly enough to ensure this happens always (or just assert this happens always from now on). Then, we can just remove the patch file, and also remove `patch-packages`, we don't need it any more (you will also need to remove a bit of code from the Generator that references this patch file).
2. Somehow ensure version of `oauth` stays 0.9.15. Then we don't need to do anything else, we leave the patch for now.
3. Try unsetting flag `--error-on-fail` (https://www.npmjs.com/package/patch-package#options-1) in `package.json`'s "postinstall" call to `patch-package`. I guess unset it by setting it to 0? Depends on how they check that flag, and if they even check it in CI. But if they do check it in CI and if they recognize 0 as false, then this should work.

**Comment by Martinsos** [comment]
What is a bigger problem here btw is that our server/ code has no package-lock.json. We should start considering how to ensure it has some kind of package-lock.json mechanism to avoid these changes in package versions all together. Maybe create another issue just for solving this.

**Comment by Martinsos** [comment]
So when using Google auth, we include `https://www.npmjs.com/package/passport-google-oauth20`. this one depends on https://www.npmjs.com/package/passport-oauth2, which depends on `oauth`.

passport-google-0auth20 allows any 1.x.x version of passport-oauth2, and newest version of passport-oauth2, released only a couple of days ago, now uses any 0.10.x version of oauth -> that is how it happened that it pulled in oauth 0.10 .

**Comment by Martinsos** [comment]
We updated this for Wasp 0.12. We decided for now not to try to fix it for 0.11.8 (by releasing 0.11.9) since 0.12 is coming out any moment. If it turns out that we have people who don't want to switch to 0.12, are not ok with monkey-patching it, and have this issue, then we can consider again releasing 0.11.9 with a fix, but I would rather we not spend resources on it at the moment.

**Comment by Martinsos** [comment]
Btw here is a "monkey-patch" fix for Wasp v0.11.8:

On your CI server, after wasp installation, add the following two commands:
```
PATCH_FILE_PATH=$(cat $(whereis wasp | cut -d " " -f 2) | tail -1 | cut -d " " -f 1 | cut -d "=" -f 2)/Generator/templates/server/package.json
sed -i 's/"postinstall": "patch-package"/"postinstall": ""/' $PATCH_FILE_PATH
```

This should prevent Wasp from trying to do a patch, therefore fixing the issue! We tested it and it works.
